driver/tests/mpi/test_restart.py

```diff
@@ -0,0 +1,188 @@
+        for var in driver_state.physics_state.__dict__.keys():
+            if isinstance(
+                driver_state.physics_state.__dict__[var],
+                gt4py.storage.storage.CPUStorage,
+            ):
+                np.testing.assert_allclose(
+                    driver_state.physics_state.__dict__[var].data,
+                    restart_physics[var].values,
+                )
+    finally:
+        subprocess.Popen(["make", "clean"])
+
+
+def test_restart_save_to_disk():
+    try:
+        backend = "gtc:numpy"
+        mpi_comm = NullComm(rank=0, total_ranks=6, fill_value=0.0)
+        partitioner = pace.util.CubedSpherePartitioner(
+            pace.util.TilePartitioner((1, 1))
+        )
+        communicator = pace.util.CubedSphereCommunicator(mpi_comm, partitioner)
+        sizer = pace.util.SubtileGridSizer.from_tile_params(
+            nx_tile=12,
+            ny_tile=12,
+            nz=79,
+            n_halo=3,
+            extra_dim_lengths={},
+            layout=(1, 1),
+            tile_partitioner=partitioner.tile,
+            tile_rank=communicator.tile.rank,
+        )
+        quantity_factory = pace.util.QuantityFactory.from_backend(
+            sizer=sizer, backend=backend
+        )
+
+        init = BaroclinicConfig()
+        driver_state = init.get_driver_state(
+            quantity_factory=quantity_factory, communicator=communicator
+        )
+        driver_state.save_state(mpi_comm)
+
+        restart_dycore = xr.open_dataset(
+            f"RESTART/restart_dycore_state_{mpi_comm.rank}.nc"
+        )
+        for var in driver_state.dycore_state.__dict__.keys():
+            if isinstance(driver_state.dycore_state.__dict__[var], pace.util.Quantity):
+                np.testing.assert_allclose(
+                    driver_state.dycore_state.__dict__[var].data,
+                    restart_dycore[var].values,
+                )
+            else:
+                if var in restart_dycore.keys():
+                    raise KeyError(
+                        f"{var} is not a quantity and \
+                        should not be in dycore restart file"
+                    )
+
+        restart_physics = xr.open_dataset(
+            f"RESTART/restart_physics_state_{mpi_comm.rank}.nc"
+        )
+        for var in driver_state.physics_state.__dict__.keys():
+            if isinstance(
+                driver_state.physics_state.__dict__[var],
+                gt4py.storage.storage.CPUStorage,
+            ):
+                np.testing.assert_allclose(
+                    driver_state.physics_state.__dict__[var].data,
+                    restart_physics[var].values,
+                )
+            else:
+                if var in restart_physics.keys():
+                    raise KeyError(
+                        f"{var} is not a storage and \
+                            should not be in physics restart file"
+                    )
+    finally:
+        subprocess.Popen(["make", "clean"])
```

Comment:
this looks like it could be dangerous - where does this run? is it always run in `driver/tests/mpi`?

I think you can add an argument to say where this command gets run, and pin it against `os.path.dirname(os.path.abspath(__file__))`.

Reply:
This will always run in `driver` because that's where the Makefile is. Alternatively, we can remove only RESTART folder and it might be less dangerous.

Reply:
What if I'm in the top level and run `pytest driver/tests/mpi`? I do this kind of command pretty regularly, especially if I'm developing more than one package.